docs/tutorials/08-tutorial-sug-roc-notleap.ipynb

```diff
@@ -0,0 +1,981 @@
+    "        volume=True,\n",
+    "        speed=True,\n",
+    "    )\n",
+    "\n",
+    "    simulation.reporters.append(dcd_reporter)\n",
+    "    simulation.reporters.append(state_reporter)\n",
+    "\n",
+    "    simulation.step(\n",
+    "        100000\n",
+    "    )  # 100 ps -- this seems to be enough to get energy convergence in each window\n",
+    "\n",
+    "    positions = simulation.context.getState(getPositions=True).getPositions()\n",
+    "\n",
+    "    with open(f\"{workdir}/{window}/production.pdb\", \"w\") as f:\n",
+    "        app.PDBFile.writeFile(model.topology, positions, f, keepIds=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "14323f00-dd6e-435e-97c1-cfe62d737e50",
+   "metadata": {},
+   "source": [
+    "#### Perform analysis"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "87f93ffb-b79f-49b6-9fd2-ed62aface6b7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import paprika.analysis as analysis\n",
+    "\n",
+    "free_energy = analysis.fe_calc()\n",
+    "free_energy.topology = \"minimized.pdb\"\n",
+    "free_energy.trajectory = \"production.dcd\"\n",
+    "free_energy.path = workdir\n",
+    "free_energy.restraint_list = guest_restraints\n",
+    "free_energy.collect_data()\n",
+    "free_energy.methods = [\"ti-block\"]\n",
+    "free_energy.ti_matrix = \"full\"\n",
+    "free_energy.boot_cycles = 1000\n",
+    "free_energy.compute_free_energy()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6477c769-efb6-4f4c-8db1-788d652ea848",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "free_energy.compute_ref_state_work(\n",
+    "    [guest_restraints[0], guest_restraints[1], None, None, guest_restraints[2], None]\n",
+    ")\n",
+    "\n",
+    "binding_affinity = -1 * (\n",
+    "    free_energy.results[\"attach\"][\"ti-block\"][\"fe\"]\n",
+    "    + free_energy.results[\"pull\"][\"ti-block\"][\"fe\"]\n",
+    "    + free_energy.results[\"ref_state_work\"]\n",
+    ")\n",
+    "\n",
+    "sem = np.sqrt(\n",
+    "    free_energy.results[\"attach\"][\"ti-block\"][\"sem\"] ** 2\n",
+    "    + free_energy.results[\"pull\"][\"ti-block\"][\"sem\"] ** 2\n",
+    ")\n",
+    "\n",
+    "result_string = f\"Binding affinity: {binding_affinity:0.4f} ± {sem:0.4f}\"\n",
+    "print(result_string)\n",
+    "with open(\"result.txt\", \"r\") as f:\n",
+    "    f.write(result_string)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9ee91f39-6968-448d-8173-38f0c14928b5",
+   "metadata": {},
+   "source": [
+    "The experimental binding free energy is -8.27 kcal/mol<sup>1</sup>. To get a more accurate calculation we can decrease the step size of the attach fractions, increase the pull distance somewhat, decrease the step size between pull distances, to name a few options. Also realize that here we are only calculating the binding of rocuronium in one orientation to one face of sugammadex (primary).\n",
```

Comment:
Do you mean to increase the number of steps (or increase sampling) in the attach phase?

Reply:
Yes, I chose to say "decreasing the step size" because my attach fractions were given by `np.arange()`, but if the attach fractions were not distributed evenly it would probably be more correct to say "increasing the number of fractions."